mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/SchedulingService.java

```diff
@@ -238,8 +238,8 @@ private TaskScheduler setupTaskSchedulerAndAutoScaler(Observable<String> vmLease
                 } else
                     logger.warn("No auto scale rules setup");
             }
-        } catch (IllegalStateException e) {
-            logger.warn("Ignoring: " + e.getMessage());
+        } catch (Exception e) {
+            logger.error("Ignoring exception during autoscale rules setup", e);
```

Comment:
@nickmahilani Maybe add a check that if this happens for all ASGs then we terminate the JVM ?

Reply:
visibility on ASG level failures is not available here, we could probably add that check in nfmantis where we get the autoscale rules to decide between retryable and non-retryable failures.

Reply:
sounds good

Reply:
I will close this PR, the getRules impl in nfmantis can throw an IllegalStateException for failures that can be ignored